front/app/api/analysis_background_tasks/useAnalysisBackgroundTask.ts

```diff
@@ -22,7 +25,19 @@ const useAnalysisBackgroundTask = (
   >({
     queryKey: backgroundTasksKeys.item({ id: backgroundTaskId }),
     queryFn: () => fetchBackgroundTask(analysisId, backgroundTaskId),
-    enabled: !!backgroundTaskId,
+    enabled: !!backgroundTaskId && !!analysisId,
+    onSuccess: () => {
+      queryClient.invalidateQueries({
+        queryKey: insightsKeys.list({ analysisId }),
+      });
+    },
+    // Refetch every 5 seconds when task is active
+    refetchInterval: (data) => {
```

Comment:
We introduce a refetch interval in order to be able to check if the task is completed.